rasa/core/tracker_store.py

```diff
@@ -527,9 +549,12 @@ def retrieve(self, sender_id):
         """
         stored = self.conversations.find_one({"sender_id": sender_id})
 
+        if not stored:
+            return
+
         # look for conversations which have used an `int` sender_id in the past
         # and update them.
-        if stored is None and sender_id.isdigit():
+        if sender_id.isdigit():
```

Comment:
this cannot be reached if sender ID is a digit this since `stored` would be `None`